ietf-wdm-tunnel.tree

```diff
@@ -3,69 +3,142 @@ module: ietf-wdm-tunnel
   augment /te:te/te:tunnels/te:tunnel:
     +--rw wdm-constraint
        +--rw transceiver-constraint
-       |  +--rw operational-modes*     string
+       |  +--rw operational-modes*            string
        |  +--rw tx-tune-constraints
-       |     +--rw min-central-frequency?    frequency-thz
-       |     +--rw max-central-frequency?    frequency-thz
-       |     +--rw transceiver-tunability?   frequency-ghz
-       +--rw gsnr-margin?                   snr
+       |  |  +--rw min-central-frequency?    frequency-thz
+       |  |  +--rw max-central-frequency?    frequency-thz
+       |  |  +--rw transceiver-tunability?   frequency-ghz
+       |  +--rw line-coding-bitrate?          identityref
+       |  +--rw tx-channel-power?             power-dbm-or-null
+       |  +--rw preferred-rx-channel-power?   l0-types:power-dbm
+       +--rw gsnr-extra-margin?             snr
        +--rw use-regen?                     boolean
        +--rw wavelength-conversion?         boolean
-       +--rw bit-stuffing?                  boolean
        +--rw wavelength-assignment?         identityref
        +--rw guard-band-size?               l0-types:frequency-thz
        +--rw matching-fwd-rev-wavelength?   boolean
        +--rw allow-retuning?                boolean
-       +--rw delta-power?                   l0-types:ratio-in-db
+       +--rw delta-power?                   l0-types:power-ratio
   augment /te:te/te:tunnels/te:tunnel/te:primary-paths
-            /te:primary-path/te:explicit-route-objects-always
+            /te:primary-path/te:explicit-route-objects
             /te:route-object-exclude-always/te:type
             /te:numbered-node-hop/te:numbered-node-hop:
     +--rw path-in-transceiver
-    |  +--rw transponder-id?        uint32
+    |  +--rw transponder-id?   uint32
```

Comment:
1. I am wondering whether we need a new hop type for the 3R function where to put this information
2. should the container be named 'path-in-transponder' and 'path-out-transponder'?
3. how to provide constraints on the transponders at the ingress/egress of the tunnel?

Reply:
Looking into an example to check if this (path-in/path-out) transponder constructs stands to describe the ERO and RROs for WDM tunnel